core/server/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1716,6 +1716,9 @@ public void resetFile(long fileId)
   public void setState(TachyonURI path, SetAttributeOptions options)
       throws FileDoesNotExistException, AccessControlException, InvalidPathException {
     MasterContext.getMasterSource().incSetStateOps(1);
+    if (options.hasOwner() || options.hasGroup() || options.hasPermission()) {
+      setAcl(path, new SetAclOptions(options));
```

Comment:
SetAclOptions? this seems messy ... I have a pending PR that introduces server side SetAttributeOptions (https://github.com/amplab/tachyon/pull/2651) and I suggest we merge that one first and the fix this up